themes/default/content/docs/troubleshooting/faq.md

```diff
@@ -89,3 +83,41 @@ $ pulumi stack import --file my-app-production.checkpoint.json # import the new
+
+To ensure that as many logical Pulumi resources and stacks can be stood up side-by-side without them conflicting, Pulumi automatically creates each resource's name by combining the logical name with a random suffix appended afterwards. This approach to auto-naming also ensures Pulumi can replace resources with less downtime, by creating the replacement before needing to delete the old resource. In practice, both of these provide significant benefits. This means, however, that your resource's physical name will typically look like `my-bucket-d7c2fa0` instead of `my-bucket`.
+
+You can disable this [auto-naming]({{< relref "programming-model" >}}#autonaming) on a per-resource basis if this isn't right for you. Doing so lets you have precise physical names without random suffixes, but also means you lose the two benefits named above.
+
+## Secrets
+
+### How does Pulumi manage secrets?
+
+When you set a [configuration]({{< relref "/docs/intro/concepts/config" >}}) value, you may pass `--secret` to `pulumi config set` which causes the value to be encrypted so it can be safely persisted in `Pulumi.<stack-name>.yaml`. For every stack the Pulumi service manages a unique encryption key, which it uses to encrypt secrets for that stack (and this is configurable to use your own custom secrets provider). Because a different key is used for each stack, encrypting the same value across two different stacks will lead to different encrypted strings being stored in the `Pulumi.<stack-name>.yaml` files. This also means that you cannot copy an encrypted value from one file to another using a text editor. Instead, you must use `pulumi config set`.
+
+When you run a preview, update or destroy, pulumi decrypts this data. It is plain text during the execution of your deployment, and any part of your Pulumi program may access it using the Pulumi config object.
+
+### Are my secrets ever visible?
+
+Pulumi provides primitives so you can enforce your [secrets]({{< relref "/docs/intro/concepts/secrets#secrets" >}}) are stored in a secure manner in the CLI, state file and Pulumi Console. During an update, your secrets will be unencrypted in memory and visible to your Pulumi program. It is your responsibility to ensure that you do not persist them outside of Pulumi without securing them.
+
+## Understanding Pulumi
+
+### Does Pulumi use Terraform?
+
+Some Pulumi providers leverage the Terraform schemas in order to know what resources and parameters are available, and to determine the return and response attributes.
+
+Pulumi does also provide native providers which are mapped directly to the cloud provider APIs and do not leverage the Terraform schemas.
```

Comment:
this is one of the questions added from Lee's blog post

Reply:
I like that we added this question. I wonder if we should be even a bit more verbose, e.g. 
```suggestion
### Does Pulumi use Terraform?

Pulumi does not use the Terraform engine in any way. However, some Pulumi providers leverage the Terraform schemas in order to know what resources and parameters are available, and to determine the return and response attributes. On the other hand, Pulumi native providers are mapped directly to the cloud provider APIs and do not leverage the Terraform schemas.
```